### PR TITLE
Add Missing Cascade Entries for planner, _llm_triage, smoke_utils, version

### LINKED ISSUE #1216: Add missing cascade entries for planner, _llm_triage, smoke_utils, version
## Summary

Four Python packages/modules under `src/autoskillit/` are missing from both `LAYER_CASCADE_CONSERVATIVE` and `LAYER_CASCADE_AGGRESSIVE` in `tests/_test_filter.py`. When any of their source files change, `build_test_scope` returns `None` (forced full test run) because `_file_to_package` returns a key not present in the cascade map.

## Implementation Order

**This is ticket 1 of 6 in the test filter improvement series. No dependencies — implement first.**

Sequence: **#1 (this)** → #2 (module-level core cascade) → #3 (tiered always-run) → #4 (coverage oracle) → #5 (diff-aware arch) → #6 (CI guard)

## Problem

At `tests/_test_filter.py` lines 682-687, any `src/` Python file whose package name is not a key in the cascade map triggers `return None` — a full test run. These four packages hit that branch:

- `planner/` — empty scaffold (`__all__ = []`), no imports, nothing imports it
- `_llm_triage.py` — imports core, execution, recipe, workspace; lazy-imported by server/helpers.py
- `smoke_utils.py` — imports core; called by string ref from recipe YAMLs
- `version.py` — no autoskillit imports; lazy-imported by server/_state.py

Note: `skills/` and `skills_extended/` are NOT affected — they contain no Python files and route through the non-Python manifest (`apply_manifest`), which already has correct entries.

### REQ-FILT-001: Add cascade entries to `LAYER_CASCADE_CONSERVATIVE`

Add before the closing `}` (currently line 189):

```python
"planner":      frozenset({"planner"}),
"_llm_triage":  frozenset({"test_llm_triage.py", "execution", "server", "recipe"}),
"smoke_utils":  frozenset({"test_smoke_utils.py", "recipe"}),
"version":      frozenset({"test_version.py", "server"}),
```

### REQ-FILT-002: Add cascade entries to `LAYER_CASCADE_AGGRESSIVE`

Add before the closing `}` (currently line 204):

```python
"planner":      frozenset({"planner"}),
"_llm_triage":  frozenset({"test_llm_triage.py"}),
"smoke_utils":  frozenset({"test_smoke_utils.py"}),
"version":      frozenset({"test_version.py"}),
```

### REQ-FILT-003: Add tests

Add parametrized test cases verifying that `build_test_scope` returns a non-None result (not a forced full run) for each of the four packages when their files appear in `changed_files`.

## Verification

```bash
python3 -c "
from tests._test_filter import build_test_scope, FilterMode
for pkg, path in [
    ('planner', 'src/autoskillit/planner/__init__.py'),
    ('_llm_triage', 'src/autoskillit/_llm_triage.py'),
    ('smoke_utils', 'src/autoskillit/smoke_utils.py'),
    ('version', 'src/autoskillit/version.py'),
]:
    result = build_test_scope({path}, FilterMode.CONSERVATIVE)
    assert result is not None, f'{pkg} still forces full run'
    print(f'{pkg}: OK ({len(result)} dirs)')
"
```

## Context

- These four packages appeared in ~25% of non-tooling integration commits
- Each forced full run costs ~183s vs the cascade-scoped run
- Report: `.autoskillit/temp/test-filter-effectiveness-report-2026-04-24.md` (Improvement 1)

### PR DESCRIPTION
**Pull request by Trecek**
## Summary

Four source packages (`planner`, `_llm_triage`, `smoke_utils`, `version`) were absent as keys in both `LAYER_CASCADE_CONSERVATIVE` and `LAYER_CASCADE_AGGRESSIVE` in `tests/_test_filter.py`. When any of their files appeared in `changed_files`, `build_test_scope` hit the `return None` fail-open branch and forced a full 183-second test run. This PR adds four entries to each cascade map and a parametrized test class verifying each package now produces a scoped result instead of `None`.

### REQ-FILT-001: Add cascade entries to `LAYER_CASCADE_CONSERVATIVE`

Add before the closing `}` (currently line 189):

```python
"planner":      frozenset({"planner"}),
"_llm_triage":  frozenset({"test_llm_triage.py", "execution", "server", "recipe"}),
"smoke_utils":  frozenset({"test_smoke_utils.py", "recipe"}),
"version":      frozenset({"test_version.py", "server"}),
```

### REQ-FILT-002: Add cascade entries to `LAYER_CASCADE_AGGRESSIVE`

Add before the closing `}` (currently line 204):

```python
"planner":      frozenset({"planner"}),
"_llm_triage":  frozenset({"test_llm_triage.py"}),
"smoke_utils":  frozenset({"test_smoke_utils.py"}),
"version":      frozenset({"test_version.py"}),
```

### REQ-FILT-003: Add tests

Add parametrized test cases verifying that `build_test_scope` returns a non-None result (not a forced full run) for each of the four packages when their files appear in `changed_files`.

### Module Dependency Diagram

```mermaid
%%{init: {'flowchart': {'nodeSpacing': 50, 'rankSpacing': 70, 'curve': 'basis'}}}%%
graph TB
    %% CLASS DEFINITIONS %%
    classDef cli fill:#1a237e,stroke:#7986cb,stroke-width:2px,color:#fff;
    classDef stateNode fill:#004d40,stroke:#4db6ac,stroke-width:2px,color:#fff;
    classDef handler fill:#e65100,stroke:#ffb74d,stroke-width:2px,color:#fff;
    classDef phase fill:#6a1b9a,stroke:#ba68c8,stroke-width:2px,color:#fff;
    classDef newComponent fill:#2e7d32,stroke:#81c784,stroke-width:2px,color:#fff;
    classDef output fill:#00695c,stroke:#4db6ac,stroke-width:2px,color:#fff;
    classDef integration fill:#c62828,stroke:#ef9a9a,stroke-width:2px,color:#fff;
    classDef terminal fill:#1a237e,stroke:#7986cb,stroke-width:2px,color:#fff;

    subgraph SRC_L0 ["LAYER 0 — FOUNDATION"]
        direction LR
        CORE["autoskillit.core/<br/>━━━━━━━━━━<br/>load_yaml · atomic_write<br/>get_logger · ClaudeFlags<br/>build_claude_env"]
    end

    subgraph SRC_L1 ["LAYER 1 — SERVICES  (4 new cascade map keys)"]
        direction LR
        PLANNER["planner/<br/>━━━━━━━━━━<br/>check_remaining<br/>build_assignment_manifest<br/>build_wp_manifest"]
        LLMTRIAGE["_llm_triage<br/>━━━━━━━━━━<br/>Contract staleness triage<br/>Haiku subprocess runner"]
        SMOKEUTILS["smoke_utils<br/>━━━━━━━━━━<br/>Smoke-test callables<br/>run_python pipeline steps"]
        VERSION["version<br/>━━━━━━━━━━<br/>Version health utilities<br/>stdlib-only · no src deps"]
        EXECUTION["execution/<br/>━━━━━━━━━━<br/>run_managed_async<br/>partition_files_by_domain<br/>annotate_diff"]
        WORKSPACE["workspace/<br/>━━━━━━━━━━<br/>bundled_skills_dir"]
    end

    subgraph SRC_L2 ["LAYER 2 — ORCHESTRATION"]
        direction LR
        RECIPE["recipe/<br/>━━━━━━━━━━<br/>StaleItem<br/>load_bundled_manifest"]
    end

    subgraph SRC_L3 ["LAYER 3 — APPLICATION"]
        direction LR
        SERVER["server/<br/>━━━━━━━━━━<br/>MCP tools<br/>imports version.py"]
    end

    subgraph TESTINFRA ["TEST INFRASTRUCTURE  (no src/ imports)"]
        direction LR
        TESTFILTER["● tests/_test_filter.py<br/>━━━━━━━━━━<br/>LAYER_CASCADE_CONSERVATIVE<br/>LAYER_CASCADE_AGGRESSIVE<br/>FilterMode · build_test_scope<br/>ASTImportWalker · check_bucket_a<br/>+4 new keys: planner _llm_triage<br/>  smoke_utils version"]
        CASCADETEST["★ tests/test_test_filter_cascade.py<br/>━━━━━━━━━━<br/>Validates cascade entries<br/>FilterMode + build_test_scope<br/>REQ-FILT-003"]
        CONFTEST["tests/conftest.py<br/>━━━━━━━━━━<br/>pytest plugin<br/>activates filter on CI"]
        TFTEST["tests/test_test_filter.py<br/>━━━━━━━━━━<br/>Primary filter test coverage<br/>(cascade tests split out)"]
    end

    %% IMPORT DEPENDENCIES (explain cascade target breadth) %%
    PLANNER -->|"imports"| CORE
    LLMTRIAGE -->|"imports"| CORE
    LLMTRIAGE -->|"imports"| EXECUTION
    LLMTRIAGE -->|"imports"| RECIPE
    LLMTRIAGE -->|"imports"| WORKSPACE
    SMOKEUTILS -->|"imports"| CORE
    SMOKEUTILS -->|"imports"| EXECUTION
    EXECUTION -->|"imports"| CORE
    WORKSPACE -->|"imports"| CORE
    RECIPE -->|"imports"| EXECUTION
    SERVER -->|"imports"| RECIPE

    %% STDLIB-only path for version %%
    VERSION -.->|"stdlib only<br/>(importlib.metadata)"| STDLIB_NODE["stdlib / importlib<br/>━━━━━━━━━━<br/>no autoskillit imports"]

    %% CASCADE MAP ENTRIES: source key → tests/_test_filter.py %%
    PLANNER ==>|"CONSERVATIVE cascade:<br/>→ planner/"| TESTFILTER
    LLMTRIAGE ==>|"CONSERVATIVE cascade:<br/>→ execution/ server/ recipe/"| TESTFILTER
    SMOKEUTILS ==>|"CONSERVATIVE cascade:<br/>→ recipe/"| TESTFILTER
    VERSION ==>|"CONSERVATIVE cascade:<br/>→ server/"| TESTFILTER

    %% TEST CONSUMER DEPENDENCIES %%
    CASCADETEST -->|"imports FilterMode<br/>build_test_scope"| TESTFILTER
    CONFTEST -->|"imports FilterMode<br/>build_test_scope<br/>git_changed_files"| TESTFILTER
    TFTEST -->|"full module import<br/>+ cascade constants"| TESTFILTER

    %% CLASS ASSIGNMENTS %%
    class CORE stateNode;
    class PLANNER,EXECUTION,WORKSPACE,VERSION,SMOKEUTILS,LLMTRIAGE handler;
    class RECIPE,SERVER phase;
    class TESTFILTER output;
    class CONFTEST,TFTEST cli;
    class CASCADETEST newComponent;
    class STDLIB_NODE integration;
```

### Scenarios Diagram

```mermaid
%%{init: {'flowchart': {'nodeSpacing': 40, 'rankSpacing': 55, 'curve': 'basis'}}}%%
flowchart LR
    %% CLASS DEFINITIONS %%
    classDef cli fill:#1a237e,stroke:#7986cb,stroke-width:2px,color:#fff;
    classDef stateNode fill:#004d40,stroke:#4db6ac,stroke-width:2px,color:#fff;
    classDef handler fill:#e65100,stroke:#ffb74d,stroke-width:2px,color:#fff;
    classDef phase fill:#6a1b9a,stroke:#ba68c8,stroke-width:2px,color:#fff;
    classDef newComponent fill:#2e7d32,stroke:#81c784,stroke-width:2px,color:#fff;
    classDef output fill:#00695c,stroke:#4db6ac,stroke-width:2px,color:#fff;
    classDef detector fill:#b71c1c,stroke:#ef5350,stroke-width:2px,color:#fff;
    classDef gap fill:#ff6f00,stroke:#ffa726,stroke-width:2px,color:#000;
    classDef terminal fill:#1a237e,stroke:#7986cb,stroke-width:2px,color:#fff;

    %% ── SCENARIO 1: Cascade Entry Test Run ────────────────────────────── %%
    subgraph S1 ["SCENARIO 1 — Cascade Entry Validation (★ New Test File)"]
        direction LR
        S1_ENTRY["★ test_test_filter_cascade.py<br/>━━━━━━━━━━<br/>TestCascadeNewEntries<br/>8 parametrized cases<br/>(planner/llm_triage/smoke_utils/version<br/>× CONSERVATIVE/AGGRESSIVE)"]
        S1_FIXTURE["tmp_path fixture<br/>━━━━━━━━━━<br/>creates fake test dirs<br/>under isolated temp root"]
        S1_BUILD["● build_test_scope()<br/>━━━━━━━━━━<br/>tests._test_filter<br/>main filter entry point"]
        S1_CASCADE["● LAYER_CASCADE_*<br/>━━━━━━━━━━<br/>planner → {planner}<br/>_llm_triage → {execution,server,recipe}<br/>smoke_utils → {recipe}<br/>version → {server}"]
        S1_ASSERT["assert result is not None<br/>━━━━━━━━━━<br/>None = full run triggered<br/>→ cascade entry missing"]
        S1_PASS["PASS<br/>━━━━━━━━━━<br/>cascade entries verified<br/>no full run for any<br/>of the 4 new packages"]
    end

    S1_ENTRY -->|"reads FilterMode<br/>build_test_scope"| S1_FIXTURE
    S1_FIXTURE -->|"passes tmp_path<br/>as tests_root"| S1_BUILD
    S1_BUILD -->|"looks up package key"| S1_CASCADE
    S1_CASCADE -->|"returns scoped dirs"| S1_ASSERT
    S1_ASSERT -->|"not None"| S1_PASS

    %% ── SCENARIO 2: Conservative Cascade Resolution ───────────────────── %%
    subgraph S2 ["SCENARIO 2 — Conservative Cascade (_llm_triage → wide fan-out)"]
        direction LR
        S2_DIFF["git diff detects<br/>━━━━━━━━━━<br/>src/autoskillit/<br/>_llm_triage.py changed"]
        S2_PKG["● _file_to_package()<br/>━━━━━━━━━━<br/>extracts '_llm_triage'<br/>from filepath"]
        S2_CONS["● LAYER_CASCADE_CONSERVATIVE<br/>━━━━━━━━━━<br/>'_llm_triage' →<br/>{test_llm_triage.py,<br/>execution, server, recipe}"]
        S2_ALWAYS["● ALWAYS_RUN_CONSERVATIVE<br/>━━━━━━━━━━<br/>+ arch, contracts,<br/>infra, docs"]
        S2_RESULT["scoped Path set<br/>━━━━━━━━━━<br/>5 dirs + always-run<br/>not None → partial run"]
    end

    S2_DIFF -->|"changed files set"| S2_PKG
    S2_PKG -->|"key lookup"| S2_CONS
    S2_CONS -->|"union"| S2_ALWAYS
    S2_ALWAYS -->|"resolve existing dirs"| S2_RESULT

    %% ── SCENARIO 3: Aggressive Cascade Resolution ─────────────────────── %%
    subgraph S3 ["SCENARIO 3 — Aggressive Cascade (_llm_triage → narrow)"]
        direction LR
        S3_DIFF["same git diff<br/>━━━━━━━━━━<br/>src/autoskillit/<br/>_llm_triage.py changed"]
        S3_AGG["● LAYER_CASCADE_AGGRESSIVE<br/>━━━━━━━━━━<br/>'_llm_triage' →<br/>{test_llm_triage.py}"]
        S3_ALWAYS["● ALWAYS_RUN_AGGRESSIVE<br/>━━━━━━━━━━<br/>+ arch, contracts only"]
        S3_RESULT["narrow Path set<br/>━━━━━━━━━━<br/>1 test file + always-run<br/>faster CI run"]
    end

    S3_DIFF -->|"same changed files"| S3_AGG
    S3_AGG -->|"union"| S3_ALWAYS
    S3_ALWAYS -->|"resolve existing files"| S3_RESULT

    %% ── SCENARIO 4: Fail-Open Guard (pre-PR baseline) ─────────────────── %%
    subgraph S4 ["SCENARIO 4 — Fail-Open Guard (what happened before this PR)"]
        direction LR
        S4_DIFF["new module changed<br/>━━━━━━━━━━<br/>src/autoskillit/<br/>planner/__init__.py<br/>(before PR: no entry)"]
        S4_PKG2["● _file_to_package()<br/>━━━━━━━━━━<br/>extracts 'planner'"]
        S4_MISS["cascade miss<br/>━━━━━━━━━━<br/>key 'planner' absent from<br/>LAYER_CASCADE_CONSERVATIVE"]
        S4_NONE["build_test_scope → None<br/>━━━━━━━━━━<br/>fail-open: full test<br/>suite triggered"]
        S4_FULL["full test run<br/>━━━━━━━━━━<br/>all test dirs executed<br/>slow CI feedback"]
    end

    S4_DIFF -->|"changed files set"| S4_PKG2
    S4_PKG2 -->|"key lookup"| S4_MISS
    S4_MISS -->|"no match → None"| S4_NONE
    S4_NONE -->|"CI runs everything"| S4_FULL

    %% ── SCENARIO 5: Planner Scoped Run (post-PR) ─────────────────────── %%
    subgraph S5 ["SCENARIO 5 — Planner Scoped Run (post-PR fix)"]
        direction LR
        S5_DIFF["src/autoskillit/<br/>planner/__init__.py<br/>━━━━━━━━━━<br/>changed (post-PR)"]
        S5_PKG3["● _file_to_package()<br/>━━━━━━━━━━<br/>extracts 'planner'"]
        S5_ENTRY2["● LAYER_CASCADE_CONSERVATIVE<br/>━━━━━━━━━━<br/>'planner' → {planner}<br/>(★ new entry added)"]
        S5_RESULT2["scoped Path set<br/>━━━━━━━━━━<br/>tests/planner/ only<br/>+ always-run dirs"]
        S5_PASS2["targeted CI run<br/>━━━━━━━━━━<br/>planner tests only<br/>fast feedback"]
    end

    S5_DIFF -->|"changed files set"| S5_PKG3
    S5_PKG3 -->|"key lookup"| S5_ENTRY2
    S5_ENTRY2 -->|"returns {planner}"| S5_RESULT2
    S5_RESULT2 -->|"not None"| S5_PASS2

    %% ── CLASS ASSIGNMENTS ── %%
    class S1_ENTRY newComponent;
    class S1_FIXTURE phase;
    class S1_BUILD,S2_PKG,S3_DIFF,S4_PKG2,S5_PKG3 handler;
    class S1_CASCADE,S2_CONS,S3_AGG,S4_MISS,S5_ENTRY2 stateNode;
    class S2_ALWAYS,S3_ALWAYS,S1_ASSERT phase;
    class S1_PASS,S2_RESULT,S3_RESULT,S5_RESULT2,S5_PASS2 output;
    class S4_NONE,S4_FULL gap;
    class S2_DIFF,S4_DIFF,S5_DIFF cli;
    class S1_ENTRY,S1_PASS newComponent;
```

Closes #1216

## Implementation Plan

Plan file: `/home/talon/projects/autoskillit-runs/impl-20260424-231056-382588/.autoskillit/temp/make-plan/add_missing_cascade_entries_plan_2026-04-24_000000.md`

🤖 Generated with [Claude Code](https://claude.com/claude-code) via AutoSkillit

## Token Usage Summary

| Step | uncached | output | cache_read | cache_write | count | time |
|------|----------|--------|------------|-------------|-------|------|
| plan | 87 | 12.1k | 308.9k | 37.8k | 1 | 5m 56s |
| verify | 132 | 15.6k | 677.5k | 48.9k | 1 | 4m 44s |
| implement | 132 | 5.3k | 609.6k | 37.6k | 1 | 1m 56s |
| fix | 158 | 7.2k | 748.4k | 42.6k | 1 | 3m 30s |
| prepare_pr | 60 | 4.3k | 183.6k | 22.7k | 1 | 1m 31s |
| run_arch_lenses | 4.3k | 16.3k | 347.7k | 90.1k | 2 | 7m 40s |
| compose_pr | 67 | 7.5k | 237.6k | 29.8k | 1 | 2m 10s |
| **Total** | 4.9k | 68.4k | 3.1M | 309.5k | | 27m 31s |

### CODE REVIEW

File: tests/test_test_filter_cascade.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        tests_root = tmp_path / "tests"
+        for d in dirs_to_create:
+            (tests_root / d).mkdir(parents=True, exist_ok=True)
+
+        result = build_test_scope(
+            changed_files={filepath},
+            mode=mode,
+            tests_root=tests_root,
+        )
+        assert result is not None, (
```
Comment (Trecek) [inline tests/test_test_filter_cascade.py:97]: [warning] tests: The assertion `result is not None` only checks that the filter does not fall back to a full run. It does not verify the actual scope content — a cascade entry that returns an unexpectedly wide or empty `set[Path]` would pass silently. Each parametrize case should additionally assert that the returned paths contain the expected cascade directories.

File: tests/test_test_filter_cascade.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            ),
+            # Aggressive: each package scoped to its own direct tests only
+            (
+                "src/autoskillit/planner/__init__.py",
+                FilterMode.AGGRESSIVE,
+                ["planner"],
+            ),
+            (
+                "src/autoskillit/_llm_triage.py",
+                FilterMode.AGGRESSIVE,
```
Comment (Trecek) [inline tests/test_test_filter_cascade.py:56]: [warning] bugs: The three aggressive-mode cases for `_llm_triage`, `smoke_utils`, and `version` all pass `dirs_to_create=[]`, so the `tests_root` directory is never created (the loop at L73-L74 does not execute). `build_test_scope` checks `is_dir()`/`is_file()` against `tests_root`; since it does not exist, every cascade entry is silently dropped and `result` is always `set()` (not `None`). The assertion `result is not None` passes vacuously — these cases do not actually validate that the cascade entries are correctly defined.

File: tests/test_test_filter_cascade.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                ["server"],
+            ),
+            # Aggressive: each package scoped to its own direct tests only
+            (
+                "src/autoskillit/planner/__init__.py",
+                FilterMode.AGGRESSIVE,
+                ["planner"],
+            ),
+            (
+                "src/autoskillit/_llm_triage.py",
```
Comment (Trecek) [inline tests/test_test_filter_cascade.py:55]: [warning] cohesion: Aggressive-mode cases for `_llm_triage`, `smoke_utils`, and `version` use `dirs_to_create=[]` while their conservative counterparts create meaningful directory trees. This asymmetry is undocumented. A comment or an assertion on what `result` contains when `tests_root` is empty would clarify intent and prevent the vacuous-pass problem.

File: tests/test_test_filter_cascade.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            (
+                "src/autoskillit/planner/__init__.py",
+                FilterMode.CONSERVATIVE,
+                ["planner"],
+            ),
+            # Conservative: _llm_triage cascades into execution/server/recipe
+            (
+                "src/autoskillit/_llm_triage.py",
+                FilterMode.CONSERVATIVE,
+                ["execution", "server", "recipe"],
```
Comment (Trecek) [inline tests/test_test_filter_cascade.py]: [info] arch: The `LAYER_CASCADE_CONSERVATIVE` entry for `_llm_triage` includes `"test_llm_triage.py"` as a direct file entry alongside `"execution"`, `"server"`, `"recipe"` directories. This test case only creates the three directories — it never creates the `test_llm_triage.py` file under `tests_root`. Verify that `build_test_scope` resolves filename entries by name (not filesystem existence) in conservative mode, or the `test_llm_triage.py` coverage branch is untested.

File: tests/_test_filter.py [hunk 5]
```diff
@@ -186,6 +186,11 @@ class ImportContext(enum.StrEnum):
             "docs",
         }
     ),
+    # Standalone modules (not subpackage directories)
+    "planner": frozenset({"planner"}),
+    "_llm_triage": frozenset({"test_llm_triage.py", "execution", "server", "recipe"}),
```
Comment (Trecek) [inline tests/_test_filter.py:191]: [info] cohesion: The `_llm_triage` entry in `LAYER_CASCADE_CONSERVATIVE` mixes a bare filename (`"test_llm_triage.py"`) with directory names (`"execution"`, `"server"`, `"recipe"`). All other entries in this dict use either directory names or plain test filenames — not both. The pattern is asymmetric compared to `smoke_utils` and `version` (filename-only) and `planner` (directory-only). A comment explaining why `_llm_triage` requires both a file and multiple directories would clarify intent.

**Review by Trecek — COMMENTED** [review]
AutoSkillit PR Review — Verdict: approved_with_comments

**Review by Trecek — COMMENTED** [review]
AutoSkillit review: warning-only findings detected. See inline comments — no blocking changes required.

File: tests/test_test_filter_cascade.py (hunk 1 above)
Comment (Trecek) [reply tests/test_test_filter_cascade.py:97]: Agreed — fixed in 2487ae03. Each parametrize case now has an expected_in_result parameter and asserts that the returned scope paths include the expected cascade directories via result_names assertions.

File: tests/test_test_filter_cascade.py (hunk 2 above)
Comment (Trecek) [reply tests/test_test_filter_cascade.py:56]: Agreed — fixed in 2487ae03. Aggressive-mode cases now create the test files via .touch() so build_test_scope can resolve them. Previously tests_root was never created, causing is_file() to return False for all entries and result to be set() rather than None — vacuous pass.

File: tests/test_test_filter_cascade.py (hunk 3 above)
Comment (Trecek) [reply tests/test_test_filter_cascade.py:55]: Agreed — fixed in 2487ae03. The asymmetry is eliminated: all cases now use items_to_create with explicit entries that match the cascade definitions, making intent clear and preventing the vacuous-pass scenario described in the related bugs comment.

File: tests/test_test_filter_cascade.py (hunk 4 above)
Comment (Trecek) [reply tests/test_test_filter_cascade.py]: Agreed — fixed in 2487ae03. Conservative cases for _llm_triage, smoke_utils, and version now also create their corresponding test files (test_llm_triage.py, test_smoke_utils.py, test_version.py) via .touch(), exercising the is_file() branch of build_test_scope (line 751 of tests/_test_filter.py) and verifying the file-based entry is included in the result.

File: tests/_test_filter.py (hunk 5 above)
Comment (Trecek) [reply tests/_test_filter.py:191]: Investigated — this is intentional. The reviewer characterizes smoke_utils and version as filename-only, but LAYER_CASCADE_CONSERVATIVE lines 192-193 show: smoke_utils: frozenset({"test_smoke_utils.py", "recipe"}) and version: frozenset({"test_version.py", "server"}). Both also mix a bare filename with a directory. The mixed pattern is symmetric and consistent across all three standalone-module entries (_llm_triage, smoke_utils, version) — no comment is needed to explain an asymmetry that does not exist. Commit 83bb3ac6 added these entries following the same pattern.
